packages/@react-spectrum/combobox/stories/ComboBox.stories.tsx

```diff
@@ -296,10 +297,42 @@ let AllControlledComboBox = (props) => {
     setInputValue(value);
   };
 
+  let setSnake = () => {
+    ReactDOM.unstable_batchedUpdates(() => {
+      setInputValue('Snake');
+      setSelectedKey('3');
+    });
+  };
```

Comment:
An example of how someone might go about updating a controlled key + controlled input value combobox without it blowing up. Alternative would be to have the user tracked state contain the key/value in a single object instead of in two separate state setters.

Reply:
I think I'm in favor of single state object with two fields, that way we don't need to use the unstable batched updates

let [fieldState, setFieldState] = useState({value: '', key: ''});

let setSnake = () => {
  setFieldState({value: 'Snake', key: '3'});
};